tsdb/engine/tsm1/compact_test.go

```diff
@@ -4218,54 +4218,136 @@ func TestEnginePlanCompactions(t *testing.T) {
 			},
 		},
 	}
-	testBlockCountsAndResults := []struct {
+
+	type testBlockCountsAndResults struct {
 		blockCounts  []int
 		level1Groups []tsm1.PlannedCompactionGroup
 		level2Groups []tsm1.PlannedCompactionGroup
 		level3Groups []tsm1.PlannedCompactionGroup
 		level4Groups []tsm1.PlannedCompactionGroup
 		level5Groups []tsm1.PlannedCompactionGroup
-	}{
+	}
+
+	type testEnginePlanCompactionsRunner struct {
+		name                      string
+		files                     []tsm1.FileStat
+		planType                  tsm1.PlanType
+		testBlockCountsAndResults testBlockCountsAndResults
+	}
+
+	tests := []testEnginePlanCompactionsRunner{
 		{
-			blockCounts: []int{
-				tsdb.DefaultAggressiveMaxPointsPerBlock,
-				tsdb.DefaultMaxPointsPerBlock,
-				tsdb.DefaultMaxPointsPerBlock,
-				tsdb.DefaultMaxPointsPerBlock,
+			name:     "test case 1 with plantype of PT_Standard",
+			files:    testFileSets[0],
+			planType: tsm1.PT_Standard,
```

Comment:
Instead of having the plantype in the test case, use nested loops. The outer loop goes through all plantypes, the inner loop goes through all test cases.

Reply:
Sounds good

Reply:
Like the tests here (though it is test cases in outer loop, indices in inner loop)

https://github.com/influxdata/influxdb/blob/eab8a8a6e83c3142d11a464df13a99b9b6bdfea5/tsdb/engine/tsm1/engine_test.go#L2362C1-L2386C2